.github/workflows/c.yaml

```diff
@@ -38,8 +38,18 @@ jobs:
           bash <(curl -s https://codecov.io/bash) -s c/build || echo "Codecov did not collect coverage reports";
 
 
-  gcc6-release:
-    name: GCC6 - Release
+  ubuntu-lts:
```

Comment:
I've applied the DRY principle here and removed the `gcc6-release` and `clang6-*` jobs and replaced them with "strategy". In github workflow you can create multiple jobs that run the same steps but with with slight differences. I've asked github to run 6 jobs here (2 were excluded, will explain this in a bit).